hr_timesheet_sheet_period/models/hr_timesheet_sheet.py

```diff
@@ -63,11 +63,29 @@ def onchange_pay_period(self, cr, uid, id, hr_period_id, context=None):
 
     def _get_current_pay_period(self, cr, uid, context=None):
         period_obj = self.pool['hr.period']
+        contract_obj = self.pool.get('hr.contract')
         date_today = datetime.today().strftime('%Y-%m-%d')
-        period_ids = period_obj.search(cr, uid,
-                                       [('date_start', '<=', date_today),
-                                        ('date_stop', '>=', date_today)],
+        employee = self.default_get(cr, uid, ['employee_id'], context=None)
+        contract_id = contract_obj.search(cr, uid,
+                                          [('employee_id', '=',
+                                            employee.get('employee_id'))],
+                                          context=context)
+        contract = contract_obj.browse(cr, uid, contract_id[0],
                                        context=context)
+        if contract.schedule_pay:
+            period_ids = period_obj.search(cr, uid,
```

Comment:
Please apply the if condition to build the search domain, and then do the search.

Something like:
```
domain = [(('date_start', '<=', date_today), ('date_stop', '>=', date_today))]
if contract.schedule_pay:
    domain += [(('schedule_pay', '=', contract.schedule_pay))]
period_ids = period_obj.search(cr, uid, search_domain, context=context)
```